src/main/java/opendota/Parse.java

```diff
@@ -201,6 +202,24 @@ public Parse(InputStream input, OutputStream output) throws IOException
       long tMatch = System.currentTimeMillis() - tStart;
       System.err.format("total time taken: %s\n", (tMatch) / 1000.0);
     }
+
+    public Parse(String input, OutputStream output, boolean quick) throws IOException, InterruptedException
```

Comment:
Make the other constructor call this one with `quick = false`; then branch in this constructor. Also, you don't have to initialize a lot of the stuff if you're doing a quick parse

```java
public Parse(InputStream input, OutputStream output) throws IOException
{
  return Parse(input, output, false); 
}

public Parse(String input, OutputStream output, boolean quick) throws IOException, InterruptedException
{
  long tStart = System.currentTimeMillis();
  if (quick) {
    CDemoFileInfo info = Clarity.infoForFile(input);
    System.out.println(info);
  } else {
    greevilsGreedVisitor = new GreevilsGreedVisitor(name_to_slot);
    trackVisitor = new TrackVisitor();
    
    is = input;
    os = output;
    isPlayerStartingItemsWritten = new ArrayList<>(Arrays.asList(new Boolean[numPlayers]));
    Collections.fill(isPlayerStartingItemsWritten, Boolean.FALSE);
    long tStart = System.currentTimeMillis();
    new SimpleRunner(new InputStreamSource(is)).runWith(this);
  }
  long tMatch = System.currentTimeMillis() - tStart;
  System.err.format("total time taken: %s\n", (tMatch) / 1000.0);
}
```